packages/css/src/generate-compression-map.ts

```diff
@@ -0,0 +1,68 @@
+import postcss from 'postcss';
+import selectorParser from 'postcss-selector-parser';
+
+import { ClassNameGenerator } from './utils/class-name-generator';
+
+const UNDERSCORE_UNICODE = 95;
+
+/**
+ * Generate a compression map, which is used by @compiled/babel-plugin to compress class names.
+ * The compression map looks like { 'aaaabbbb': 'a', 'bbbbcccc': 'b' }
+ *
+ * @param stylesheet css content i.e. `.aaaabbbb{font-size: 10px}`
+ * @param oldClassNameCompressionMap the previous compression map, which ensures the compression is deterministic.
+ * @returns newClassNameCompressionMap
+ */
+export const generateCompressionMap = (
+  css: string,
+  oldClassNameCompressionMap?: { [index: string]: string }
+): undefined | { [index: string]: string } => {
+  let classNamesToCompress: string[] = [];
+  const classNameCompressionMap: { [index: string]: string } = {};
+  const reservedClassNames = ['ad', 'ads']; // Prevent AdBlocker from removing the class names
```

Comment:
Nice! 

Out of curiosity, was this based on Tom's comment or did you find any resource with a list of AdBlocker classes.
Maybe `adv` could be an issue too ( [eg](https://stackoverflow.com/a/20242141))? 

Also, given that we prefix classes with `_`, I wonder if compressed classes are safe, regardless

Reply:
Good point! I had a look at the easylist. I think I will take the approach of [styled-components](https://github.com/styled-components/styled-components/blob/main/packages/styled-components/src/utils/generateAlphabeticName.ts#L1), which avoids any word includes `ad`. Let me update the code.

Reply:
Should we add 'ddb' to this list as well? [EasyList](https://easylist.to/easylist/easylist.txt) (which is available for Adblock Plus, uBlock Origin etc) will block any class named "ddb"

> Also, given that we prefix classes with _, I wonder if compressed classes are safe, regardless

I'm curious about this too :)